document_session_cluster_transaction.go

```diff
@@ -0,0 +1,328 @@
+package ravendb
+
+import (
+	"reflect"
+)
+
+type ClusterTransactionOperations struct {
+	session                             *InMemoryDocumentSessionOperations
+	state                               map[string]*CompareExchangeSessionValue
+	missingDocumentsTooAtomicGuardIndex map[string]string
+}
+
+func (cto *ClusterTransactionOperations) GetNumberOfTrackedCompareExchangeValues() int {
+	if nil == cto.state {
+		return 0
+	}
+
+	return len(cto.state)
+}
+
+func (cto *ClusterTransactionOperations) TryGetMissingAtomicGuardFor(docId string) (*string, bool) {
+	if cto.missingDocumentsTooAtomicGuardIndex == nil {
+		return nil, false
+	}
+
+	value, exists := cto.missingDocumentsTooAtomicGuardIndex[docId]
+	if exists == false {
+		return nil, false
+	}
+
+	return &value, true
+}
+
+func (cto *ClusterTransactionOperations) IsTracked(key string) bool {
+	_, exists := cto.tryGetCompareExchangeValueFromSession(key)
```

Comment:
Shouldn't we already check `missingDocumentsTooAtomicGuardIndex` ?

Reply:
Well, I accidentally put it (I was using v5.2 JVM as reference implementing this but sometimes I checked C# client). I will skip `missingDocumentsToAtomicGuardIndex` for now and leave it for future development.